codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ScalaGeneratorUtils.scala

```diff
@@ -81,6 +93,45 @@ object ScalaGeneratorUtils {
     if (types.isEmpty) ""
     else types.mkString("[", ", ", "]")
 
+  def dataTypeCodeBlock(typeArgument: ModelBuilder.TypeArgument): CodeBlock =
+    typeArgument match {
+      case ModelBuilder.MessageTypeArgument(messageType) =>
+        c"$messageType"
+      case ModelBuilder.ScalarTypeArgument(scalar) =>
+        scalar match {
+          case ModelBuilder.ScalarType.Int32 | ModelBuilder.ScalarType.UInt32 | ModelBuilder.ScalarType.SInt32 |
+              ModelBuilder.ScalarType.Fixed32 | ModelBuilder.ScalarType.SFixed32 =>
+            c"Int"
+          case ModelBuilder.ScalarType.Int64 | ModelBuilder.ScalarType.UInt64 | ModelBuilder.ScalarType.SInt64 |
+              ModelBuilder.ScalarType.Fixed64 | ModelBuilder.ScalarType.SFixed64 =>
+            c"Long"
+          case ModelBuilder.ScalarType.Double => c"Double"
+          case ModelBuilder.ScalarType.Float  => c"Float"
+          case ModelBuilder.ScalarType.Bool   => c"Boolean"
+          case ModelBuilder.ScalarType.String => c"String"
+          case ModelBuilder.ScalarType.Bytes  => c"ByteString"
+          case _                              => c"_"
+        }
+    }
+
+  def parameterizeDataTypeCodeBlock(replicatedData: ModelBuilder.ReplicatedData): CodeBlock =
+    replicatedData match {
+      // special case ReplicatedMap as heterogeneous with ReplicatedData values
+      case ModelBuilder.ReplicatedMap(key) => c"[${dataTypeCodeBlock(key)}, ReplicatedData]"
+      case data =>
+        val types = data.typeArguments.map(typ => dataTypeCodeBlock(typ))
+        if (types.isEmpty) c""
+        else {
+          var sep = c""
+          val separatedTypes = types.foldLeft(c"") { (acc, el) =>
+            val cur = c"$acc$sep$el"
+            sep = c", "
+            cur
+          }
+          c"[$separatedTypes]"
+        }
+    }
```

Comment:
This is a duplication for `dataType` and `parameterizeDataType` that works with `CodeBlock` instead of String and implicit imports. 

The replicated entity generator is the only Scala generator that is not using codeblocks. Probably because of the lack of those methods. We can use it as a migration path to introduce CodeBlock here as well. 

Last but not least I added this dirty hack at the end that is basically a `mkString` for CodeBlocks. Can certainly be improved.